### DISCUSSION TITLE
How to change the itemLabel of Collection Items?

### DISCUSSION THREAD

**Discussion by srotread** [Thinkmill/keystatic#374] [Q&A]
Like the array field, where we can specify an itemLabel. Is there something similar for Collection items? Currently, it uses the slug.
![image](https://github.com/Thinkmill/keystatic/assets/136879970/ab6b14a6-99bc-42d8-af94-1c872424e552)

Even if it defaults to the slug field, it would be awesome if it could use the regular text version rather than the slugified version.
![image](https://github.com/Thinkmill/keystatic/assets/136879970/7824b276-40a2-4bca-bdc1-bfcee7c79fd3)

Maybe I am missing something and this already exists but you can't pass a second object into the collection function like so `collection({ }, { })` and the first object doesn't have any itemLabel field.

**Comment by JedWatson** [comment]
Related to the comment I just left on #375 

💯 agree showing the label (or another field) here would be nicer, but it requires us to read the contents of each file, right now it's showing the filenames (the entry slug) which is significantly more performant at scale (noting that unless you're running in local mode, we're loading data over the GitHub API)

We'll do something about it soon, just need to make sure we don't blow things up in the process!

**Comment by simrikservices** [comment]
Just a thought, as a temporary solution, why not just replace "-" with SPACE and and use title casing ?

Ex: the-slug-of-a-post  = The Slug of a Post.
